calculators/_memoria_rating.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+$GLOBALS['rating_calculators']['_memoria_rating'] = [
+	'description' => '
+		by <a href="/player/125980">Memoria</a>: Rating with per-matchup rating calculation, v1
+		
+		-Only for 1v1, 2v2, and 3v3.
+  		-Games under 4 minutes don\'t count.
+		-Just like in the main rating, people start at 2000 rating,
+		 4000 rating difference = 90% win prediction, max 100 rating per match
+  		-Rating prediction is calculated for each matchup (1v1),
+		 then it\'s combined into a final prediction 
+		',
+	'default_rating' => 2000,
+	'fields' => [
+		'g.duration',
+	],
+	'main' => '_memoria_rating__main',
+	];
+
+function _memoria_rating__main(&$teams) {
+	$game_duration_seconds = floor($teams[0][0]['duration']/22.4);
+	if (count($teams[0]) != count($teams[1])) {
+		return;
+	}
+	
+	if ($game_duration_seconds < 240) {
+		return;
+	}
+		
+	$player_c = count($teams[0]);
+	$factor = 4000;
+	$ratings1 = [];
+	foreach ($teams[0] as $player) {
+        $ratings1[] = $player['rating'];
+    }
+	$ratings2 = [];
+	foreach ($teams[1] as $player) {
+        $ratings2[] = $player['rating'];
```

Comment:
this operation is a waste of not only CPU, but the person reading it brain resources.
The only place you are using it is in ($ratings1[$i]-$ratings2[$i]), I don't see much difference with ($teams[0][$i]['rating']-$teams[1][$i]['rating'])

up to you, but I highly recommend